### PR TITLE
8316197: Make tracing of inline cache available in unified logging

### PR DESCRIPTION
**Pull request by yftsai**
This removes develop flag `TraceICs` and makes the logs available via `-Xlog`.

Example:
```
% java -Xlog:inlinecache=trace -version
[0.061s][trace][inlinecache] CompiledDirectStaticCall@0x00007f3739001d17: set_to_interpreted java.lang.StringLatin1.hashCode([B)I
[0.078s][trace][inlinecache] IC@0x00007f3739004a87: monomorphic to compiled (rcvr klass = nullptr)
[0.079s][trace][inlinecache] CompiledDirectStaticCall@0x00007f3739005dff: set_to_interpreted jdk.internal.util.ArraysSupport.vectorizedHashCode(Ljava/lang/Object;IIII)I
[0.079s][trace][inlinecache] CompiledDirectStaticCall@0x00007f373900502f: set_to_interpreted jdk.internal.org.objectweb.asm.ByteVector.enlarge(I)V
[0.079s][trace][inlinecache] IC@0x00007f373900502f: monomorphic to interpreter: {method} {0x00007f36f03e6318} 'enlarge' '(I)V' in 'jdk/internal/org/objectweb/asm/ByteVector'
[0.079s][trace][inlinecache] CompiledDirectStaticCall@0x00007f3739006b0f: set_to_compiled 0x00007f3739002120
[0.083s][trace][inlinecache] CompiledDirectStaticCall@0x00007f373900928f: set_to_interpreted java.lang.AbstractStringBuilder.newCapacity(I)I
[0.083s][trace][inlinecache] IC@0x00007f373900928f: monomorphic to interpreter: {method} {0x00007f36f00cd170} 'newCapacity' '(I)I' in 'java/lang/AbstractStringBuilder'
...
```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8316197](https://bugs.openjdk.org/browse/JDK-8316197): Make tracing of inline cache available in unified logging (**Enhancement** - P4)

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [5d782f75](https://git.openjdk.org/jdk/pull/17026/files/5d782f759d357dc3d49bfd0c11806ae0c096ec8b)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17026/head:pull/17026` \
`$ git checkout pull/17026`

Update a local copy of the PR: \
`$ git checkout pull/17026` \
`$ git pull https://git.openjdk.org/jdk.git pull/17026/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17026`

View PR using the GUI difftool: \
`$ git pr show -t 17026`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17026.diff">https://git.openjdk.org/jdk/pull/17026.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17026#issuecomment-1846249050)

### CODE REVIEW

**Review by vnkozlov — CHANGES_REQUESTED** [review]
You missed `compiler/arguments/TestTraceICs.java` test.

File: test/hotspot/jtreg/compiler/arguments/TestTraceICs.java [hunk 1]
```diff
@@ -25,8 +25,7 @@
  * @test
  * @bug 8217447
  * @summary Test running TraceICs enabled.
- * @run main/othervm -XX:+IgnoreUnrecognizedVMOptions -XX:+TraceICs
- *                   compiler.arguments.TestTraceICs
+ * @run main/othervm compiler.arguments.TestTraceICs
```
Comment (vnkozlov) [inline test/hotspot/jtreg/compiler/arguments/TestTraceICs.java:28]: I think you need to add `-Xlog:inlinecache=trace` flag to keep previous test's behavior and check your new code.

File: test/hotspot/jtreg/compiler/arguments/TestTraceICs.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
  * @test
  * @bug 8217447
  * @summary Test running TraceICs enabled.
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/compiler/arguments/TestTraceICs.java:27]: The summary is no longer applicable.

But really this test seems some what pointless. It basically checks that `Xlog:inlinecache` is a valid log setting, but doesn't actually check anything interesting.

**Review by dholmes-ora — APPROVED** [review]
Code changes look fine to me.

I think the test can go but that is up to @vnkozlov .

Thanks

File: test/hotspot/jtreg/compiler/arguments/TestTraceICs.java (hunk 2 above)
Comment (yftsai) [reply test/hotspot/jtreg/compiler/arguments/TestTraceICs.java]: Updated. TestLogJIT is similar in only enabling the log setting without checking anything.

Comment (vnkozlov) [reply test/hotspot/jtreg/compiler/arguments/TestTraceICs.java]: Yes, these test could check output but it is separate issue. Let keep them and update later.

**Review by dholmes-ora — APPROVED** [review]

**Pull request closed without merging**